src/components/MsgPanels.js

```diff
@@ -140,11 +141,20 @@ export function ConnectionStatus(props) {
 }
 
 export function ErrorDetails(props) {
-    const { header = "Error details:", style = { fontSize: "0.8em", overflow: "auto" } } = props;
-    return (
+    const { header = "Error details:", style = { fontSize: "0.8em", overflow: "auto" }, details } = props;
+
+    return !details ? null : (
         <div>
             <p>{header}</p>
-            <pre style={style}>{props.children}</pre>
+
+            {details && <pre style={style}>{details.message ? details.message : details}</pre>}
+
+            {/* {details.stack (
```

Comment:
why are there commented lines?

Reply:
stack was quite unhelpful to display. content of it is in the message anyway (eg. try to reject tx signature in metamask to test)
we will need to tweak this error handling in the future, hide the sacry details by default and handle basic scenarios (e.g signature rejection, metamask is locked etc.). I think we've a trello card for it.